core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -996,6 +996,21 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       }
     }
 
+    // Delete log files that don't exist in listing database and exceed the configured max age.
+    Option(fs.listStatus(new Path(logDir))).map(_.toSeq).getOrElse(Nil)
+      .flatMap { entry => EventLogFileReader(fs, entry) }
+      .foreach { reader =>
+        try {
+          listing.read(classOf[LogInfo], reader.rootPath.toString)
+        } catch {
+          case _: NoSuchElementException =>
+            if (reader.modificationTime < maxTime) {
```

Comment:
If this is a `time-based` deletion for unlisted logs, why don't we have at line 980?

Reply:
Thank you for review, @dongjoon-hyun 

You're right, let me update.

Reply:
> If this is a `time-based` deletion for unlisted logs, why don't we have this at line 980?

@dongjoon-hyun , have updated, could you help follow up this PR, thanks.